src/iroh.rs

```diff
@@ -474,159 +422,281 @@ impl VeilidIrohBlobs {
 
         return Ok(read);
     }
-    pub async fn create_collection(&self, collection_name: String) -> anyhow::Result<Hash> {
-        
+    pub async fn create_collection(&self, collection_name: &String) -> anyhow::Result<Hash> {
         println!("Creating a new empty collection with name: {}", collection_name);
-        
-        // Create a new empty collection
-        let collection = Collection::default();
     
-         // Log: Serializing the collection to CBOR
-        println!("Serializing the collection to CBOR...");
+        // Create a new empty HashMap for the collection
+        let collection: HashMap<String, Hash> = HashMap::new();
     
-        // Serialize the collection to CBOR
+        // Serialize the HashMap to CBOR
         let cbor_data = to_vec(&collection)?;
-
-        // Log: Writing CBOR data to a temporary file
-        println!("Writing CBOR data to a temporary file...");
-        
-        // Create a temporary file to store the CBOR data
+    
+        // Write the CBOR data to a temporary file
         let temp_path = self.base_dir.join(format!("{}_collection.cbor", collection_name));
         std::fs::write(&temp_path, &cbor_data)?;
-        
-        // Convert the path to an absolute path
+    
+        // Convert the path to an absolute path and upload it to the store
         let absolute_temp_path = std::fs::canonicalize(&temp_path)?;
-
-        // Log: Uploading the CBOR data from file to the store
-        println!("Uploading the CBOR data from file to the store...");
-
-        // Upload the CBOR data to the store using upload_from_path
         let collection_hash = self.upload_from_path(absolute_temp_path).await?;
```

Comment:
Before merging we should make sure we upload from stream instead

Reply:
Switched to upload_from_stream as part of the encapsulated method -> `update_collection`

Reply:
b3fbc4ddf3cab612a93f5f9eef78632ab131cd99